### PR TITLE
[SOLR-17368] Change prometheus dummy metric to a more unique label

### PR DESCRIPTION
**Pull request by mlbiscoc**
https://issues.apache.org/jira/browse/SOLR-17368

# Description

`TestPrometheusResponseWriter` was failing ~25% of tests due to dummy metric `dummyMetrics.2xx-responses` not unique enough for exporting compared to `org.eclipse.jetty.server.handler.DefaultHandler.2xx-responses` which could leak between tests creating a duplicate metric.

# Solution

Change the dummy jetty metric with `999xx` that exports a unique enough label that `2xx` cannot clash with it, even with real leaking jetty metrics.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by mlbiscoc** [comment]
@dsmiley This should make the dummy metric unique enough to not clash with real leaking jetty metrics from other tests. I ran tests locally multiple times. Any ideas what we can do to make sure that is resilient enough? I want to avoid breaking CI and generating noise for a 3rd time...

File: solr/core/src/test/org/apache/solr/response/TestPrometheusResponseWriter.java [hunk 1]
```diff
@@ -78,7 +78,12 @@ public void testPrometheusStructureOutput() throws Exception {
           output.lines().filter(line -> !line.startsWith("#")).collect(Collectors.toList());
       filteredResponse.forEach(
           (actualMetric) -> {
-            String actualValue = actualMetric.substring(actualMetric.lastIndexOf("} ") + 1);
+            String actualValue;
+            if (actualMetric.contains("}")) {
+              actualValue = actualMetric.substring(actualMetric.lastIndexOf("} ") + 1);
+            } else {
+              actualValue = actualMetric.split(" ")[1];
+            }
```
Comment (mlbiscoc) [inline solr/core/src/test/org/apache/solr/response/TestPrometheusResponseWriter.java:86]: Also adding this in to better parse numeric value out of prometheus metrics based on whether or not tags exist

**Comment by dsmiley** [comment]
Were you able to reproduce the problem?  If I understand the cause, you'd need to run a number of tests, not just this one test.
The root fix would be to stop metric leaks between tests!  Albeit not today -- okay.

**Comment by mlbiscoc** [comment]
Yes, on `main` branch it fails the same way when running Jetty based tests next to it. From Chris's suggestion, `./gradlew -p solr/core test -Ptests.jvms=1 -Ptests.verbose=true --tests TestPrometheusResponseWriter --tests \*Jetty\*` Ran is multiple times and this looks to clean up the issue on my branch.

Current test failure looks to be from `LocalFSCloudIncrementalBackupTest`? Don't think this is related to this though.

**Pull request merged**